addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactBroadcastListener.java

```diff
@@ -0,0 +1,196 @@
+            }
+
+            try {
+                broadcastChannel.close();
+            } catch (IOException e) {
+                logger.error("An exception occurred while closing the broadcast channel on port number {} ({})",
+                        LISTENER_PORT_NUMBER, e.getMessage());
+            }
+
+            isStarted = false;
+        }
+    }
+
+    public void registerHandler(KeContactHandler handler) {
+
+        if (handler != null) {
+            listeners.add(handler);
+
+            if (logger.isTraceEnabled()) {
+                logger.trace("There are now {} Keba handlers registered with the broadcast listener", listeners.size());
+            }
+
+            if (listeners.size() == 1) {
+                start();
+            }
+        }
+
+    }
+
+    public void unRegisterHandler(KeContactHandler handler) {
+        if (handler != null) {
+            listeners.remove(handler);
+
+            if (logger.isTraceEnabled()) {
+                logger.trace("There are now {} Keba handlers registered with the broadcast listener", listeners.size());
+            }
+
+            if (listeners.size() == 0) {
+                stop();
+            }
+        }
+    }
+
+    public Runnable broadcastRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                ByteBuffer readBuffer = null;
+                InetSocketAddress clientAddress = null;
+
+                SelectionKey theSelectionKey = broadcastChannel.keyFor(broadcastListener);
+
+                if (theSelectionKey != null) {
+
+                    synchronized (broadcastListener) {
+                        try {
+                            broadcastListener.selectNow();
+                        } catch (IOException e) {
+                            logger.error("An exception occurred while selecting: {}", e.getMessage());
+                        }
+                    }
+
+                    Iterator<SelectionKey> it = broadcastListener.selectedKeys().iterator();
+                    while (it.hasNext()) {
+                        SelectionKey selKey = it.next();
+                        it.remove();
+                        if (selKey.isValid() && selKey.isReadable() && selKey == theSelectionKey) {
+
+                            readBuffer = ByteBuffer.allocate(KeContactHandler.BUFFER_SIZE);
+                            int numberBytesRead = 0;
+
+                            if (selKey == broadcastKey) {
+                                try {
+                                    clientAddress = (InetSocketAddress) broadcastChannel.receive(readBuffer);
+                                    logger.debug("Received {} from {} on the broadcast listener port ",
+                                            new String(readBuffer.array()), clientAddress);
+                                    numberBytesRead = readBuffer.position();
+
+                                } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

In this case is it maybe sufficient to catch the exception (or some of them) in the outer try-catch block?

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception